automation/all_clippy_checks.sh

```diff
@@ -17,10 +17,20 @@ fi
 
 EXTRA_ARGS=( "$@" )
 
+# set this in the environment for the script when we want to use the
+# experimental automated clippy fixes from a future version of nightly
+# to ease the pain of doing an upgrade:
+if [[ -z ${CLIPPY_FIX+x} ]]
+then
+    CLIPPY_HEAD="cargo clippy"
+else
+    CLIPPY_HEAD="cargo +nightly clippy --fix -Z unstable-options"
```

Comment:
Neat! I guess there's a chance that this could apply fixes from nightly clippy, that stable clippy disagrees with...but we can grapple with those if/when they come up.